src/travel-details-screens/__tests__/has_short_wait_time_and_not_guaranteed_correspondance.test.ts

```diff
@@ -115,17 +127,119 @@ describe('hasShortWaitTimeAndNotGuaranteedCorrespondence', () => {
         expectedStartTime: '2024-10-31T15:06:00Z',
         expectedEndTime: '2024-10-31T15:20:00Z',
       },
+    ] as Leg[];
+
+    expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(true);
+  });
+
+  it('should be false despite short wait time between second and third leg, beacuse interchange is guarenteed first leg and mode equals foot in second leg.', () => {
+    const legs = [
+      {
+        expectedStartTime: '2024-10-31T14:45:00Z',
+        expectedEndTime: '2024-10-31T15:00:00Z',
+        interchangeTo: {guaranteed: true},
+      },
       {
         mode: 'foot',
-        expectedStartTime: '2024-10-31T15:25:00Z',
-        expectedEndTime: '2024-10-31T15:30:00Z',
+        expectedStartTime: '2024-10-31T15:00:00Z',
+        expectedEndTime: '2024-10-31T15:05:00Z',
       },
       {
-        expectedStartTime: '2024-10-31T15:31:00Z',
-        expectedEndTime: '2024-10-31T15:40:00Z',
+        expectedStartTime: '2024-10-31T15:06:00Z',
+        expectedEndTime: '2024-10-31T15:20:00Z',
       },
     ] as Leg[];
 
     expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(false);
   });
+
+  it('should be true because because of short wait time between second and third leg, and the mode in the second leg is not foot.', () => {
+    const legs = [
+      {
+        expectedStartTime: '2024-10-31T14:45:00Z',
+        expectedEndTime: '2024-10-31T15:00:00Z',
+        interchangeTo: {guaranteed: true},
+      },
+      {
+        mode: 'bus',
+        expectedStartTime: '2024-10-31T15:00:00Z',
+        expectedEndTime: '2024-10-31T15:05:00Z',
+      },
+      {
+        expectedStartTime: '2024-10-31T15:06:00Z',
+        expectedEndTime: '2024-10-31T15:20:00Z',
+      },
+    ] as Leg[];
+
+    expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(true);
+  });
+
+  it('should be true because of short wait time and no guarenteed correspondance.', () => {
+    const legs = [
+      {
+        mode: 'foot',
+        expectedStartTime: '2024-10-31T14:45:00Z',
+        expectedEndTime: '2024-10-31T15:00:00Z',
+      },
+      {
+        expectedStartTime: '2024-10-31T15:00:00Z',
+        expectedEndTime: '2024-10-31T15:05:00Z',
+      },
+    ] as Leg[];
+
+    expect(hasShortWaitTimeAndNotGuaranteedCorrespondence(legs)).toBe(true);
+  });
```

Comment:
I think this one is wrong, and the test should be 'should be false when short wait time after first foot leg' or something like it. 

When there is just walking as the first leg, the journey planner algorithm always make it be a the bus stop "just in time". This means that with this logic, we would get the short wait time message on every trip starting with a foot leg.

Reply:
Okey, I´ll update the logic👍